File: acceptance/sulabhghimire/roman_to_int.dart
class Solution {
  List<int> values = [1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1];
  List<String> symbol = [
    "M",
    "CM",
    "D",
    "CD",
    "C",
    "XC",
    "L",
    "XL",
    "X",
    "IX",
    "V",
    "IV",
    "I"
  ];
  int romanToInt(String s) {
    int total = 0;

    for (int i = 0; i < s.length; i++) {
      try {
        if (values[symbol.indexOf(s[i])] >= values[symbol.indexOf(s[i + 1])]) {
          total = total + values[symbol.indexOf(s[i])];
        } else {
          total = total - values[symbol.indexOf(s[i])];
        }
      } finally {
        if (i == (s.length - 1)) {
          total = total + values[symbol.indexOf(s[i])];
          break;
        }
      }
    }
    return total;
  }
}

File: acceptance/arjun/romantoint.py
class Solution(object):
    
    def romanToInt(self, st):
        number = 0
        equivalent = {"I":1,"V":5,"X":10,"L":50,"C":100,"D":500,"M":1000}
        for i in range(len(st)-1):
                if (equivalent[st[i]] < equivalent[st[i+1]]):
                    number -= equivalent[st[i]]
                else:
                    number += equivalent[st[i]]
        
        return number+equivalent[st[-1]]